File: data/derived_cot/rq1_traces/translation/qwen/java_to_cpp/code_output/SplitSentence.cpp
#include <vector>
#include <string>
#include <regex>
#include <sstream>
#include <cctype>

class SplitSentence {
public:
    std::vector<std::string> splitSentences(const std::string& sentencesString) {
        if (sentencesString.empty()) {
            return {};
        }

        std::regex pattern(R"( (?<!\w\.\w.) (?<![A-Z][a-z]\.) (?<=[\.\?])\s )");
        auto begin = std::sregex_iterator(sentencesString.begin(), sentencesString.end(), pattern);
        auto end = std::sregex_iterator();

        std::vector<std::string> sentences;
        size_t lastEnd = 0;
        for (auto i = begin; i != end; ++i) {
            std::smatch match = *i;
            sentences.push_back(sentencesString.substr(lastEnd, match.start() - lastEnd));
            lastEnd = match.end();
        }
        if (lastEnd < sentencesString.length()) {
            sentences.push_back(sentencesString.substr(lastEnd));
        }
        return sentences;
    }

    int countWords(const std::string& sentence) {
        if (sentence.empty()) {
            return 0;
        }

        std::string cleaned;
        for (char c : sentence) {
            if (std::isalnum(c) || c == ' ') {
                cleaned += c;
            }
        }

        std::istringstream iss(cleaned);
        std::string word;
        int count = 0;
        while (iss >> word) {
            count++;
        }
        return count;
    }

    int processTextFile(const std::string& sentencesString) {
        std::vector<std::string> sentences = splitSentences(sentencesString);
        int maxCount = 0;
        for (const auto& sentence : sentences) {
            int count = countWords(sentence);
            if (count > maxCount) {
                maxCount = count;
            }
        }
        return maxCount;
    }
};

File: data/derived_cot/rq3_prompting/results/translation/qwen/java_to_py/code_output/SplitSentence.py
import re

class SplitSentence:
    @staticmethod
    def split_sentences(sentences_string):
        pattern = r"(?<!\w\.\w.)(?<![A-Z][a-z]\.)(?<=\.|\?|\!)\s"
        matches = re.finditer(pattern, sentences_string)
        last_end = 0
        sentences = []
        for match in matches:
            start = match.start()
            end = match.end()
            sentences.append(sentences_string[last_end:end-1])
            last_end = end
        if last_end < len(sentences_string):
            sentences.append(sentences_string[last_end:])
        return sentences

    @staticmethod
    def count_words(sentence):
        cleaned_sentence = re.sub(r'[^a-zA-Z\s]', '', sentence)
        words = cleaned_sentence.split()
        return len(words)

    @staticmethod
    def process_text_file(sentences_string):
        sentences = SplitSentence.split_sentences(sentences_string)
        max_count = 0
        for sentence in sentences:
            count = SplitSentence.count_words(sentence)
            if count > max_count:
                max_count = count
        return max_count